rust/src/lib/ovn.rs

```diff
@@ -0,0 +1,268 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Deserializer, Serialize};
+use std::collections::HashMap;
+use std::fmt;
+use std::str::FromStr;
+
+pub const OVN_BRIDGE_MAPPINGS: &str = "ovn-bridge-mappings";
+
+#[derive(Debug, Clone, PartialEq, Eq, Default, Serialize)]
+#[non_exhaustive]
+pub struct OvnConfiguration {
```

Comment:
Public document required. Try `/// `

Reply:
Done - I pretty much just moved it from the `OvnBridgeMapping` struct to here. 

Let me know if we need more docs.